docs/commerce/payment/extend_payment.md

```diff
@@ -0,0 +1,120 @@
+---
+description: Extend Payment with custom payment method type.
+edition: commerce
+---
+
+# Extend Payment
+
+There are different ways you can extend your Payment module implementation. 
+One of them is to create a custom payment method type. 
+The other is attaching custom data to a payment.
+
+You can also [customize the payment processing workflow](configure_payment.md#custom_payment_workflows).
+
+## Create custom payment method type
+
+If your application needs payment methods of other type than the default `offline` one, you must create custom payment method types. 
+
+!!! note "Gateway integration requirement"
+
+    [[= product_name =]] does not come with gateway redirects. Whether you are an integrator or the end customer, it is your responsibility to implement payment gateway integration.
+
+### Define custom payment method type
+
+Create a PHP definition of the payment method type.
+
+``` php
+[[= include_file('code_samples/front/shop/payment/src/PaymentMethodType/PayPal/PayPal.php') =]]
+```
+
+Make sure that `getName` returns a human readable name of the payment method type, the way you want it to appear on the list of available payment method types.
+
+Now, register the definition as a service:
+
+``` yaml
+[[= include_file('code_samples/front/shop/payment/config/packages/services.yaml', 0, 5) =]]
+
+```
+
+As an alternative, instead of creating a PHP library, you can use the Type Factory to define the payment method type in your YAML configuration file:
+
+``` yaml
+services:
+    app.payment.type.paypal: 
+        class: Ibexa\Contracts\Payment\PaymentMethod\Type\TypeInterface
+        factory: [ '@Ibexa\Contracts\Payment\PaymentMethod\Type\TypeFactoryInterface', 'createType' ]
+        arguments: 
+            $identifier: 'paypal' 
+            $name: 'PayPal' 
+        tags: 
+            - name: ibexa.payment.payment_method.type 
+              alias: paypal
+
+```
+
+At this point a custom payment method type should be visible in the user interface.
```

Comment:
Will it be visible at this point, before you add forms? Or only visible in the payment list, but not when you click it?